app/models/request/ChannelOptionalRequest.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.request
+
+import play.api.mvc.WrappedRequest
+
+trait ChannelOptionalRequest[A] extends WrappedRequest[A] {
+
+  def getChannel: String = headers.get("channel") match {
+    case Some(channel) => channel
+    case None          => "web"
+  }
```

Comment:
I don't think we should have any default, if the header isn't specified then it should be a Bad Request.

Since the FE hasn't implemented the work to incorporate with this, this is a buildup of technical debt.

In the mean time, while we live with the TD, this isn't adding an _optional_ field of channel, so I don't think that this name is correct